src/render/tests/test_accel_bypass.py

```diff
@@ -0,0 +1,214 @@
+    """Rendering with and without acceleration data structures should result in the same images."""
+    # Adapted from test_renders.test_render()
+    significance_level = 0.01
+    resx, resy = (103, 51)
+
+    # Compute spp budget
+    sample_budget = int(1e6)
+    pixel_count = resx * resy
+    spp = sample_budget // pixel_count
+
+    results = {}
+    for bypass in (True, False):
+        scene = make_test_scene(resx, resy, use_naive_intersection=bypass)
+        # Render the scene, including a variance estimate
+        scene.integrator().render(scene, seed=0, spp=spp)
+
+        bmp = scene.sensors()[0].film().bitmap(raw=False)
+        img, var_img = bitmap_extract(bmp, require_variance=True)
+        results[bypass] = (img, var_img)
+
+    # Compute Z-test p-value
+    p_value = z_test(results[0][0], spp, results[1][0], results[1][1])
+
+    # Apply the Sidak correction term, since we'll be conducting multiple independent
+    # hypothesis tests. This accounts for the fact that the probability of a failure
+    # increases quickly when several hypothesis tests are run in sequence.
+    alpha = 1.0 - (1.0 - significance_level) ** (1.0 / pixel_count)
+
+    success = (p_value > alpha)
+    if (np.count_nonzero(success) / 3) >= (0.9975 * pixel_count):
+        print(f'Accepted the null hypothesis (min(p-value) = {np.min(p_value)}, '
+              f'significance level = {alpha})')
+    else:
+        print(f'Rejected the null hypothesis (min(p-value) = {np.min(p_value)}, '
+              f'significance level = {alpha})')
+
+        # Note: images are in the XYZ color space by default
+        for bypass in results:
+            xyz_to_rgb_bmp(results[bypass][0]).write(
+                f'test_{mi.variant()}_{"naive" if bypass else "accel"}.exr')
+
+        assert False, 'Z-test failed'
+
+
+# Useful to investigate performance, but probably not reliable enough
+# to run on the CI.
+@pytest.mark.skip
+@fresolver_append_path
+def test02_speed(variants_all_backends_once):
+    log_level = mi.Thread.thread().logger().log_level()
+    mi.set_log_level(mi.LogLevel.Warn)
+
+    results = {}
+    for bypass in (True, False):
+        results[bypass] = []
+        scene = make_test_scene(
+            512, 256, simple=True,
+            integrator={'type': 'direct'},
+            use_naive_intersection=bypass)
+
+        for i in range(12):
+            dr.eval()
+            dr.sync_thread()
+            t0 = time.time()
+            img = mi.render(scene, spp=32)
+            dr.eval(img)
+            dr.sync_thread()
+
+            if i >= 2:
+                elapsed = time.time() - t0
+                results[bypass].append(elapsed)
+
+            if i == 0:
+                mi.Bitmap(img).write(f'test_speed_{mi.variant()}_{bypass}.exr')
+
+    print(f'\n--- {mi.variant()} ---')
+    for bypass, values in results.items():
+        print(f'{"naive" if bypass else "accel"}: {1000 * np.mean(values):.6f} ms')
```

Comment:
Currently getting these results for a scene containing a single analytic sphere, with the `direct` integrator:

```
--- scalar_rgb ---
naive: 71.434951 ms
[0.06400513648986816, 0.06670022010803223, 0.07629013061523438, 0.07175850868225098, 0.06787610054016113, 0.08331823348999023, 0.06868195533752441, 0.0772852897644043, 0.07218050956726074, 0.0662534236907959]
accel: 73.885083 ms
[0.07530069351196289, 0.07167458534240723, 0.07322049140930176, 0.06887698173522949, 0.07498621940612793, 0.08003997802734375, 0.07488870620727539, 0.0752708911895752, 0.07259535789489746, 0.07199692726135254]
------------------

--- llvm_ad_rgb ---
naive: 51.368260 ms
[0.04977750778198242, 0.055547237396240234, 0.04845452308654785, 0.04771161079406738, 0.05972123146057129, 0.04860043525695801, 0.0483553409576416, 0.059114694595336914,
0.047663211822509766, 0.0487368106842041]
accel: 51.119304 ms
[0.0474705696105957, 0.05318307876586914, 0.0463564395904541, 0.04656219482421875, 0.061937808990478516, 0.05034995079040527, 0.046396732330322266, 0.04927563667297363, 0.06153106689453125, 0.04812955856323242]
------------------

--- cuda_ad_rgb ---
naive: 13.897085 ms
[0.014396190643310547, 0.013736724853515625, 0.013788223266601562, 0.013866901397705078, 0.013941049575805664, 0.014055013656616211, 0.013768434524536133, 0.013695955276489258, 0.013820171356201172, 0.01390218734741211]
accel: 12.030840 ms
[0.012321233749389648, 0.01208186149597168, 0.011997222900390625, 0.011968374252319336, 0.01200246810913086, 0.011963605880737305, 0.011999368667602539, 0.01200246810913086, 0.011951684951782227, 0.012020111083984375]
------------------
```

This doesn't include the savings from accel build up and updates, e.g. in the context of an optimization loop.
Also, the case where I saw huge gains before was with a hardcoded axis-aligned cube intersection routine. Currently our `cube` shape is a type of `Mesh`, so we'd need a new shape type to find out if it reproduces.

Reply:
The output values of the time measurement?

If you're talking about the rendered images, they're really close, but if the intersections are computed in different ways, then I don't find it surprising that they are small imprecisions?

Reply:
ah, are those things in brackets also time values? 🤦 

Regarding the actually computed intersections: differences around the machine epsilon (relative error ~1e-07) would be expected.

Reply:
Okay yeah we're not at machine epsilon :p
To be investigated, it might be that the implementations actually compute slightly different results (not just due to e.g. order of operations).